sw/device/lib/dif/dif_csrng.c

```diff
@@ -36,6 +113,68 @@ dif_csrng_result_t dif_csrng_configure(const dif_csrng_t *csrng,
   return kDifCsrngOk;
 }
 
+dif_csrng_result_t dif_csrng_instantiate(
+    const dif_csrng_t *csrng, dif_csrng_entropy_src_toggle_t entropy_src_enable,
+    const dif_csrng_seed_material_t *seed_material) {
+  const csrng_app_cmd_t app_cmd = {
+      .id = kCsrngAppCmdInstantiate,
+      .entropy_src_enable = entropy_src_enable,
+      .seed_material = seed_material,
+      .generate_len = 0,
+  };
+  return write_application_command(csrng, &app_cmd);
+}
+
+dif_csrng_result_t dif_csrng_reseed(
+    const dif_csrng_t *csrng, const dif_csrng_seed_material_t *seed_material) {
+  const csrng_app_cmd_t app_cmd = {
+      .id = kCsrngAppCmdReseed,
+      .entropy_src_enable = false,
+      .seed_material = seed_material,
+      .generate_len = 0,
+  };
+  return write_application_command(csrng, &app_cmd);
+}
+
+dif_csrng_result_t dif_csrng_update(
+    const dif_csrng_t *csrng, const dif_csrng_seed_material_t *seed_material) {
+  const csrng_app_cmd_t app_cmd = {
+      .id = kCsrngAppCmdUpdate,
+      .entropy_src_enable = false,
+      .seed_material = seed_material,
+      .generate_len = 0,
+  };
+  return write_application_command(csrng, &app_cmd);
+}
+
+dif_csrng_result_t dif_csrng_generate(const dif_csrng_t *csrng, size_t len) {
+  if (len == 0) {
+    return kDifCsrngBadArg;
+  }
+
+  // Round up the number of 128bit blocks. Aligning with respect to uint32_t.
+  uint32_t num_128bit_blocks = len >> 2;
+  num_128bit_blocks += (len & 0x3) ? 1 : 0;
```

Comment:
I've seen this kind of  "round up to alignment" or similar stuff a lot. Do we want to start a base/math.h? @cfrantz

Reply:
For what it's worth, the usual way to write this sort of thing is `(len + 3) / 4` (because `floor((N + (k - 1)) / k) = ceil(N/k)`).

Reply:
Added an issue and associated TODO re. math.h, and updated code to match @rswarbrick's suggestion.